RecoMuon/MuonIdentification/src/MuonTimingFiller.cc

```diff
@@ -94,11 +94,27 @@ MuonTimingFiller::fillTiming( const reco::Muon& muon,
   if ( !(muon.combinedMuon().isNull()) ) {
     theDTTimingExtractor_->fillTiming(dtTmSeq, muon.combinedMuon(), iEvent, iSetup);
     theCSCTimingExtractor_->fillTiming(cscTmSeq, muon.combinedMuon(), iEvent, iSetup);
-  } else
+  } else {
     if ( !(muon.standAloneMuon().isNull()) ) {
       theDTTimingExtractor_->fillTiming(dtTmSeq, muon.standAloneMuon(), iEvent, iSetup);
       theCSCTimingExtractor_->fillTiming(cscTmSeq, muon.standAloneMuon(), iEvent, iSetup);
+    } else {
+      if ( muon.isTrackerMuon() ) {
+	std::vector<const DTRecSegment4D*> dtSegments;
+	std::vector<const CSCSegment*> cscSegments;
+	for( auto& chamber: muon.matches() ){
+	  for ( auto& segment : chamber.segmentMatches ){
```

Comment:
are these all arbitrated already at this point?
were there plots posted already that this simple selection gives good results?

Reply:
These are all matched segments. I think it's a good point. In some cases segments that belong to different muons where one is in time and the other is not can mess up the timing measurement and make it less useful. Since the arbitration information is already available at the point when we fill timing information I'll add it.